femtools/VTK_interfaces.F90

```diff
@@ -238,6 +241,7 @@ subroutine vtk_write_fields(filename, index, position, model, sfields,&
     type(tensor_field), dimension(:), intent(in), optional :: tfields
     logical, intent(in), optional :: write_region_ids
     logical, intent(in), optional :: write_columns
+    type(scalar_field), dimension(:), optional :: metadata
```

Comment:
Hm, this seems to abuse scalar_field as a container for a real array (with a name). A scalar_field is defined on a mesh (function space!), and it suggests we care which mesh. I can see this is a compact way of supplying one or more real arrays of different lengths with names, but if people want to use this in a generic way they really will have to abuse scalar_fields by allocating a %val array that is inconsistent with whatever random mesh they choose to use. Also when first reading this code, I thought it was just writing out an extra field (i.e an actual spatial field), so I think it's better to make it explcit that we're just supplying a real array of arbitrary length.

Since we only use this for time at the moment, I would be happy to just have a `real, intent(in):: time_value` argument (where the name is hardcoded to `TimeValue`). Alternatively you could use an array of `real_vector`s (see femtools/Futils.F90) with a separate array of names.